File: app/channels/posts_channel.rb
class PostsChannel < ApplicationCable::Channel
  def subscribed
    stream_for "room_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def receive(data)
    #to do create message 
    PostsChannel.broadcast_to("room_channel", data)
  end
end

File: app/javascript/channels/posts_channel.js
import consumer from "channels/consumer"

export default  consumer.subscriptions.create("PostsChannel", {
  connected() {
    console.log("connected")
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    //console.log(data)
    const rootElement = document.querySelector("#messages")
    var element = document.createElement('p')
    element.innerText = data.message
    rootElement.append(element);
    // Called when there's incoming data on the websocket for this channel
  }

 
});



